ocean-components/src/main/java/br/com/useblu/oceands/components/compose/OceanInputSearch.kt

```diff
@@ -0,0 +1,270 @@
+                unfocusedContainerColor = containerColor,
+                focusedTextColor = activeColor,
+                unfocusedTextColor = inactiveColor,
+                focusedLeadingIconColor = activeSearchIcon,
+                unfocusedLeadingIconColor = inactiveColor,
+                focusedPlaceholderColor = inactiveColor,
+                unfocusedPlaceholderColor = inactiveColor,
+            ),
+            leadingIcon = leadingIconSetup(
+                isFocused = isFocused,
+                input = input.text,
+            ),
+            trailingIcon = trailingIconSetup(
+                input = input.text,
+                onClick = {
+                    input = input.copy(text = "")
+                }
+            ),
+            placeholder = placeholderSetup(
+                isFocused = isFocused,
+                input = input.text,
+                placeholder = placeholder
+            ),
+        )
+    }
+}
+
+@Composable
+fun createOnValueChangeHandler(
+    value: TextFieldValue,
+    inputType: OceanInputType,
+    onTextChanged: (String) -> Unit
+): (TextFieldValue) -> Unit {
+
+    var input by remember { mutableStateOf(value) }
+
+    return { newValue: TextFieldValue ->
+        val modifiedValue = inputType.modifyBeforeOnChange(newValue.text)
+        if (modifiedValue != input.text) {
+            onTextChanged(modifiedValue)
+            input = newValue.copy(text = modifiedValue)
+            if (inputType.alwaysGoToEndOfInput()) {
+                input = input.copy(
+                    selection = TextRange(modifiedValue.length)
+                )
+            }
+        } else {
+            input = newValue.copy(text = modifiedValue)
+        }
+    }
+}
+
+@Composable
+fun trailingIconSetup(
+    input: String,
+    onClick: () -> Unit
+): @Composable (() -> Unit)? {
+    val activeColor = OceanColors.interfaceDarkUp
+
+    return if (input.isNotEmpty()) {
+        {
+            Icon(
+                modifier = Modifier.clickable { onClick.invoke() },
+                painter = painterResource(id = OceanIcons.X_OUTLINE.icon),
+                tint = activeColor,
+                contentDescription = null,
+            )
+        }
+    } else null
+}
+
+@Composable
+fun leadingIconSetup(
+    isFocused: Boolean,
+    input: String,
+): @Composable (() -> Unit) {
+    val activeColor = OceanColors.brandPrimaryPure
+    val inactiveColor = OceanColors.interfaceLightDeep
+
+    return {
+        if (isFocused || input.isNotEmpty()) {
+            Icon(
+                painter = painterResource(id = OceanIcons.SEARCH_OUTLINE.icon),
+                tint = activeColor,
+                contentDescription = null,
+            )
+        } else {
+            Icon(
+                painter = painterResource(id = OceanIcons.SEARCH_OUTLINE.icon),
+                tint = inactiveColor,
+                contentDescription = null,
+            )
+        }
+    }
```

Comment:
aqui como a única coisa que muda no if else, acredito que fica mais simples se fizer algo tipo:

```suggestion
    val tintColor = if (isFocused || input.isNotEmpty()) {
        activeColor
    } else inactiveColor
    return {
        Icon(
            painter = painterResource(id = OceanIcons.SEARCH_OUTLINE.icon),
            tint = tintColor,
            contentDescription = null,
        )
    }
```